src/middlewares.rs

```diff
@@ -6,29 +6,49 @@ use axum::{
 };
 use axum_extra::extract::CookieJar;
 
-#[derive(Clone, Debug)]
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct Course {
+    pub name: String,
+    pub crns: Vec<String>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct UserState {
-    pub course_reg_numbers: Vec<String>,
+    pub selection: Vec<Course>,
 }
 
-pub type CookieUserState = Extension<Option<UserState>>;
+pub type CookieUserState = Extension<UserState>;
 
 pub async fn parse_cookie(
     cookie: CookieJar,
     mut req: Request,
     next: Next,
 ) -> Result<Response, (StatusCode, String)> {
-    let user_state = if let Some(user_state) = cookie.get("state") {
+    let user_state = if let Some(raw_state) = cookie.get("state") {
         // parse the cookie here
         // TODO: currently (of my expectation) the cookie contains the
         // comma seperated CRN's. Need to query Malcolm's scraped data
         // for whatever attributes needed for course display.
-        dbg!(&user_state);
-        let course_reg_numbers: Vec<String> = Vec::new();
-        Some(UserState { course_reg_numbers })
+        dbg!(&raw_state);
+        let blank_selection: Vec<Course> = Vec::new();
+        UserState {
+            selection: blank_selection,
+        }
     } else {
-        None
+        let mut blank_selection: Vec<Course> = Vec::new();
+        blank_selection.push(Course {
+            name: "csc111".to_string(),
+            crns: Vec::new(),
+        });
+        UserState {
+            selection: blank_selection,
+        }
     };
+
+    dbg!(&serde_json::to_string(&user_state).unwrap());
```

Comment:
we can now json serialize the userstate and store it in the `state` cookie, then json deserialize it when we get a request

Reply:
could also json serialize -> base64 encode -> put in cookie